### DISCUSSION TITLE
When does alpine initialise

### DISCUSSION THREAD

**Discussion by nathanmerryInvestoo** [alpinejs/alpine#605] [4. General]
Hi, 

I am currently using Alpine with blade and it is working great.

I am wondering when alpine.js runs and loads on the components that are created (through blade),

for example does it all fire after the document has loaded.

I want to be sure that there is no render blocking.

Thanks in advance

**Comment by ryangjchandler** [comment]
@nathanmerryInvestoo everything happens on `DOMContentLoaded`, you can see here: https://github.com/alpinejs/alpine/blob/a0fab06379b5427df82a8a32e495e373f51c7a16/src/index.js#L11

As for render blocking, best thing to do is put it at the end of your `<body>` tag without a `defer` attribute, or make sure the `defer` attribute it added when the `<script>` is inside of the `<head>`, as with any asset.

**Reply by nathanmerryInvestoo** [reply]
awesome, thank you! :)
